### ISSUE TITLE
How to print the mean IoU during training

### ISSUE THREAD

**Issue by John1231983** [zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow#20]
Hello, I would like to add one thing in your training code. During training, I want to print mean IoU (besides step and loss). I add in your model.py in line [262](https://github.com/zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow/blob/master/model.py#L262)  

```
 #mIoU
pred_logits = tf.reshape(self.pred, [-1, ])
gt = tf.reshape(self.label_batch, [-1, ])
 # Ignoring all labels greater than or equal to n_classes.
temp = tf.less_equal(gt, self.conf.num_classes - 1)
weights = tf.cast(temp, tf.int32)
# fix for tf 1.3.0
gt = tf.where(temp, gt, tf.cast(temp, tf.uint8))        
 self.mIoU, self.mIou_update_op = tf.contrib.metrics.streaming_mean_iou(pred_logits, gt, num_classes=self.conf.num_classes, weights=weights)
```

Line [39](https://github.com/zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow/blob/master/model.py#L39)
```
self.sess.run(tf.local_variables_initializer())
```
Line [53](https://github.com/zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow/blob/master/model.py#L53)

```
                loss_value, images, labels, preds, summary, _,_, = self.sess.run(
                    [self.reduced_loss,
                    self.image_batch,
                    self.label_batch,
                    self.pred,
                    self.total_summary,
                    self.train_op,
                    self.mIou_update_op],
                    feed_dict=feed_dict)
m_IoU = self.mIoU.eval(session=self.sess)
```
And line [68](https://github.com/zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow/blob/master/model.py#L68)

 ```
print('step {:d} \t loss = {:.3f}, ({:.3f} sec/step), Mean IoU: {:.3f}'.format(step, loss_value, duration, m_IoU))
```
But the result of mIoU did not change among steps. Do you know what is reason and how could I fix it? Thanks
```
step 0 	 loss = 4.039, (3.927 sec/step), Mean IoU: 0.004
step 1 	 loss = 3.142, (0.893 sec/step), Mean IoU: 0.004
step 2 	 loss = 2.124, (0.660 sec/step), Mean IoU: 0.004
step 3 	 loss = 1.549, (0.655 sec/step), Mean IoU: 0.004
step 4 	 loss = 2.705, (0.677 sec/step), Mean IoU: 0.004
step 5 	 loss = 1.956, (0.700 sec/step), Mean IoU: 0.004
...
step 502 	 loss = 0.769, (0.852 sec/step), Mean IoU: 0.004
step 503 	 loss = 0.525, (0.831 sec/step), Mean IoU: 0.004
step 504 	 loss = 1.202, (0.812 sec/step), Mean IoU: 0.004
```

**Comment by John1231983** [comment]
Sorry. I was wrong to add mIoU in the condition. It must be add in the else condition
